### PR TITLE
🎨 Using shared analysis options for all packages

### PR DESCRIPTION
**Pull request by AlexV525**
Uses the base analysis options for all packages to avoid redundant definitions.

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

**Comment by AlexV525** [comment]
Unfortunately, I don't have ideas about how to test compute with these changes.

**Comment by kuhnroyal** [comment]
It may be better to just use https://pub.dev/packages/lints and adjust a few rule that we want additionally or don't want? This way new rules would be used automatically.

**Comment by AlexV525** [comment]
> It may be better to just use https://pub.dev/packages/lints and adjust a few rule that we want additionally or don't want? This way new rules would be used automatically.

That's for sure, but it cannot be used in a base config like this, since we don't have a pubspec here. Otherwise we may revert the config change and change dart files only in this PR.

**Comment by AlexV525** [comment]
I walked around other monorepos like fluttefire, flutter_local_notification, and melos itself, adding a pubspec seems fine here.

**Comment by ueman** [comment]
We could symlink the same file to all repos. Then we could use imports. I am not sure whether that’s a good idea.

**Comment by AlexV525** [comment]
> We could symlink the same file to all repos. Then we could use imports. I am not sure whether that’s a good idea.

We should avoid symlinks because it usually breaks the development on Windows. Added new commits.

File: pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: dio_workspace
```
Comment (AlexV525) [inline pubspec.yaml:1]: - https://github.com/invertase/melos/blob/main/pubspec.yaml
- https://github.com/firebase/flutterfire/blob/master/pubspec.yaml

Comment (kuhnroyal) [reply pubspec.yaml:1]: Yea they all do this because they use `melos`. But I think it is a good idea.

Comment (kuhnroyal) [reply pubspec.yaml:1]: But this involves us changing Github workflows to run `pub get` in the root repository before anything else.

**Comment by kuhnroyal** [comment]
Tbh I think we can just add `lints` to every package. There should not be that much customization of rules required.

**Comment by AlexV525** [comment]
> But this involves us changing Github workflows to run `pub get` in the root repository before anything else.

I'm surprised not seing any exceptions in tests. Maybe it only affects the local development.

> Tbh I think we can just add `lints` to every package. There should not be that much customization of rules required.

That's what we have done previously.

**Comment by AlexV525** [comment]
> It works perfectly fine without the `pubspec.yaml` in root for me.

This is what happened if you don't have a pubspec in the root which doesn't have `.dart_tool` too.
image

**Comment by kuhnroyal** [comment]
> > It works perfectly fine without the `pubspec.yaml` in root for me.
> 
> This is what happened if you don't have a pubspec in the root which doesn't have `.dart_tool` too. image

Yea but this is only when you open it right? We can just name it differently and it is probably fine. Something like `analysis_options_base.yaml` might work.

**Comment by AlexV525** [comment]
Deleted and scheduled for another run.

**Review by chenglu — APPROVED** [review]
LGTM!

**Pull request merged**